api/queries_sponsors.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import "net/http"
+
+type SponsorQuery struct {
```

Comment:
I totally agree with your point, here. I was having some typing issues with your suggested implementation, so I decided to punt that refactor for later. The code snippet you've provided should accelerate me to that refactor 🙏 I'll dive back into this a bit after satisfying the rest of the AC for the first issue

Reply:
> However, I could also make an argument that not reusing the same type in the implementation and test is actually a good thing because it makes it less likely that the same mistake in structure appears in both places.

🤔 a bit more about this and I might argue this is overkill, because you've got another place that does this check: the mocked data. GraphQL won't work correctly if the shape of the mocked data doesn't match the expected response interface, so really this means we have 3 things that must line up among these types for checking. This feels like overkill, so I could argue that reusing this interface between the code and tests to avoid this unpleasant construction in tests is more valuable than having a 3rd error check.

Reply:
It's definitely the less valuable part in comparison to seeing the data shape at one glance. To avoid the gross construction I would lean towards having the test server return stringified JSON directly. This also has a minor benefit of clearly documenting what we should get back over the wire from the server.

With regards the server type checking, yes, absolutely. I'm not really describing increased confidence in final correctness, I'm speaking more to the development process. Like, I can run a query, grab the real JSON from the server, drop it into the test server response and go from there.

Reply:
It's all tradeoffs though, of course. I just think over time I've landed on this as the most comfortable for me.

Reply:
Also, it seems the union type is required:

> GraphQL: Selections can't be made directly on unions (see selections on Sponsor) (query SponsorsList.user.sponsors.nodes.login)

I think this means I can't reuse the struct for the Reflection method vs the String Manipulation method (or I haven't been able to figure out how to reuse it, yet).

It seems the String Manipulation method allows a clear way to define that `login` can come from an organization or a user and the response body doesn't care about either. However, I suspect that the Reflection method requires the response to include both the Organization and User fields in the response and then map them accordingly, meaning that the []Nodes{} structs are different:

**String Manipulation**
```golang
type Node struct {
    Login string `json:"login"`
}
```

**Reflection**
```golang
type Node struct {
    Organization struct {
        Login string `json:"org_login" graphql:"... on Organization`
    }
    User struct {
        Login string `json:"user_login" graphql:"... on User"`
    }
}
``` 

My gut says that String Manipulation is the better choice, here, because it utilizes the graphql BE to do the mapping of Org and User logins to a single list, rather than doing it locally.

That said, I'm probably missing something 😅